### PR TITLE
Move Release Manager responsibilities to release-managers doc

### PR DESCRIPTION
**Pull request by xmudrii**
#### What type of PR is this:

/kind cleanup

#### What this PR does / why we need it:

We have responsibilities and expectations for Release Managers and Release Manager Associates listed in three places:

- `release-managers.md` document
- Branch Management handbook
- Patch Release Team handbook

The purpose of this PR is to collect responsibilities and expectations and document them in a single place. I've chosen the `release-managers.md` document to be the place where we put that information, but if there's a better place for this, I'd be happy to update the PR.

#### Which issue(s) this PR fixes:

Relevant to #1324

#### Special notes for your reviewer:

I have two concerns regarding this PR for which I'm looking for feedback:

- Does this accurately reflect what was in the handbooks?
- Does this accurately reflect the work that Release Managers and Release Manager Associates are currently doing?

/assign @saschagrunert @justaugustus @cpanato @hasheddan @puerco 
cc: @kubernetes/release-engineering @kubernetes/sig-release-leads

### CODE REVIEW

**Comment by xmudrii** [comment]
/hold
for discussion

File: release-managers.md
Comment (hasheddan) [inline release-managers.md:36]: Should we add a note here that patch release team / branch manager used to be separate responsibilities but they are now under the umbrella of a release manager? It could be a bit confusing to see the different "team" documents and assume that not all release managers are part of those teams.

Comment (xmudrii) [reply release-managers.md:36]: @hasheddan That’s a good point, I agree that it could be a bit confusing. I’ll add a note.

Comment (xmudrii) [reply release-managers.md:36]: @hasheddan I've added a note.

File: release-managers.md
Comment (saschagrunert) [inline release-managers.md:61]: I think we can remove this line, it's not intended to use `krel ff` any more.

**Review by saschagrunert — COMMENTED** [review]
Just a nit, otherwise LGTM :point_down:

**Comment by gianarb** [comment]
I think it contains the information I expect and I like that it is shorter than the old one and centralized.
Well done

/approve

File: release-managers.md
Comment (cpanato) [reply release-managers.md:61]: same as sascha said. 
lgtm

Comment (xmudrii) [reply release-managers.md]: Done.

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by mkorbi** [comment]
thank you @xmudrii looks good!
/lgtm

**Comment by saschagrunert** [comment]
@xmudrii feel free to lift the hold if this is ready to merge.

**Comment by xmudrii** [comment]
Thank y'all for the reviews!
/hold cancel

**Pull request merged**
